packages/storage-plus/src/indexed_snapshot.rs

```diff
@@ -59,7 +60,7 @@ impl<'a, K, T, I> IndexedSnapshotMap<'a, K, T, I> {
 impl<'a, K, T, I> IndexedSnapshotMap<'a, K, T, I>
 where
     T: Serialize + DeserializeOwned + Clone,
-    K: PrimaryKey<'a> + Prefixer<'a>,
+    K: PrimaryKey<'a> + Prefixer<'a> + KeyDeserialize,
```

Comment:
Do you need to add this for everything?

I would keep the bounds as they are, and then add a block with this extra bound for the functions that actually deserialise the key. From the diff, you aren't using `K: KeyDeserialize` yet

I had a number of longer discussions with Bart especially about only declaring bounds we really need, and not including them on the struct, but rather the impl (and using multiple impl blocks if we only need a given bound for a few methods.

Reply:
Ah, I realize later that this called `deserialize_kv` and although you didn't change this code, the implementation of `deserialize_kv` has changed to require this bound. 

The above comment is most likely obsolete.

Reply:
Exactly. This is because of the default deserialization we're now using. We can improve on that, by creating a series of extra constructors in `Prefix`, like `with_raw_key`. And then we can then segregate the required trait bounds. Will try to do it.